cloud/pkg/controller/controller/downstream.go

```diff
@@ -212,13 +213,33 @@ func (dc *DownstreamController) syncEdgeNodes(stop chan struct{}) {
 								break
 							}
 							msg.BuildRouter(constants.EdgeControllerModuleName, constants.GroupResource, resource, model.UpdateOperation)
-							msg.Content = dc.lc.GetAllServices()
+							svcs := dc.lc.GetAllServices()
+							msg.Content = svcs
 							if err := dc.messageLayer.Send(*msg); err != nil {
 								log.LOGGER.Warnf("Send message failed with error: %s, operation: %s, resource: %s", err, msg.GetOperation(), msg.GetResource())
 							} else {
 								log.LOGGER.Infof("Send message successfully, operation: %s, resource: %s", msg.GetOperation(), msg.GetResource())
 							}
 
+							for _, svc := range svcs {
```

Comment:
This function has too many lines, split this function into multiple sub-functions according to the function please.